include/onnxruntime/core/providers/dml/dml_provider_factory.h

```diff
@@ -117,15 +117,16 @@ struct OrtDmlApi {
 
   /**
    * Creates a DirectML Execution Provider using the given DirectML device, and which executes work on the supplied D3D12
-   * command queue. The DirectML device and D3D12 command queue must have the same parent ID3D12Device, or an error will
-   * be returned. The D3D12 command queue must be of type DIRECT or COMPUTE (see D3D12_COMMAND_LIST_TYPE). If this
+   * command queue. If a DirectML device isn't provided, a default DirectML device will be created with the same D3D12 device
+   * that the supplied command queue is using. The DirectML device and D3D12 command queue must have the same parent ID3D12Device,
+   * or an error will be returned. The D3D12 command queue must be of type DIRECT or COMPUTE (see D3D12_COMMAND_LIST_TYPE). If this
    * function succeeds, the inference session maintains a strong reference on both the dml_device and the command_queue
    * objects.
    * See also: DMLCreateDevice
    * See also: ID3D12Device::CreateCommandQueue
    */
   ORT_API2_STATUS(SessionOptionsAppendExecutionProvider_DML1, _In_ OrtSessionOptions* options,
-                  _In_ IDMLDevice* dml_device, _In_ ID3D12CommandQueue* cmd_queue);
+                  _In_opt_ IDMLDevice* dml_device, _In_ ID3D12CommandQueue* cmd_queue);
```

Comment:
>_In_opt_  [](http://example.com/codeflow?start=18&length=9)

[nit] If sal annotation automatically passes `nullptr` to the optional parameter, then should we keep the optional parameter at the end?